File: AutomaticDrawing.Core/Core/Painter/EventArgs/UpdatePaintEventArgs.vb
''' <summary>
''' 为绘制更新时的事件提供数据
''' </summary>
Public Class UpdatePaintEventArgs
    ''' <summary>
    ''' 新的绘制点
    ''' </summary>
    Public Property Point As Vertex
    ''' <summary>
    ''' 绘制进度
    ''' </summary>
    Public Property Percent As Single
    ''' <summary>
    ''' 创建并初始化一个实例
    ''' </summary>
    Public Sub New(point As Vertex, percent As Single)
        Me.Point = point
        Me.Percent = percent
    End Sub
End Class

File: AutoPaint.Core/Painter/EventArgs/UpdatePaintEventArgs.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AutoPaint.Core
{
    public class UpdatePaintEventArgs
    {
        /// <summary>
        /// Gets the vertex being draw.
        /// </summary>
        public Vertex NewVertex { get; }

        /// <summary>
        /// Gets the progress of painting.
        /// </summary>
        public float Percent { get; }

        public UpdatePaintEventArgs(Vertex point, float percent)
        {
            this.NewVertex = point;
            this.Percent = percent;
        }
    }
}
